### ISSUE TITLE
Monitoring network form - modal for deleting timeseries from network

### ISSUE THREAD

**Issue by lexvand** [nens/lizard-management-client#713]

**Comment by Derryrover** [comment]
@LirryPinter 
@lexvand 

Hi Lirry and Lex,

We already had a discussion concerning the timeseries being added to the monitoring network in this ticket:
https://github.com/nens/lizard-management-client/issues/654

We concluded, by Lex his latest comment, that the existing "multiple select dropdown" is not sufficient for adding timeseries to monitoring networks, because it does not handle large amount of timeseries well.

Anyway, originally we had also another multiple select field called the "slushbucket".
It can still be seen here in the rasterlayer-design for "share with organisations".
https://xd.adobe.com/view/eda684bb-be65-4b2b-879f-fc6b96240847-b8f5/screen/97b3722c-f6d3-4193-b58e-5570d5cf8192

Would this field maybe allow us to have all in one field; adding; deleting; viewing , timeseries for a monitoring network?
Alternatively we could add more search functionality and columns to this field.

I propose this alternative, because I think that having separate screens for deleting and adding timeseries to monitoring networks is not the ideal situation. It also requires yet another custom field (instead of re-using existing ones).

Please let me know what you think.

kr Tom

**Comment by lexvand** [comment]
@Derryrover Good to have an example of another implementation. I see the benefits of this approach.

A few questions:
- Does this solution support filtering on location and parameter on both buckets?
- Can it be based on backend pagination or would you first need to retrieve all objects and work with frontend mechanisms to keep track of the changes? Can this become problematic for large datasets?

**Comment by Derryrover** [comment]
> * Does this solution support filtering on location and parameter on both buckets?
> 

The existing slushbucket component currently does not support searching on the right-side (selection) bucket.
This type of features could be added.
I think also that, additionally to the filter/search, the user should also be able to just scroll/paginate trough all the timeseries that are part of the monitoringnetwork. Namely, it can be that the user does not know name or details of the timeseries she is looking for. Or maybe she simply wants to have a list of all timeseries or remove them all from the monitoring network. This also impacts your second question.

> * Can it be based on backend pagination or would you first need to retrieve all objects and work with frontend mechanisms to keep track of the changes? Can this become problematic for large datasets?

Large datasets can become a problem yes. I see this as a legitimate concern, but if I am correct then the current design also has this problem. The slushbucket component at the moment requires to first retrieve all items. It is possible to develop it further so it can dynamically load available timeseries from the api. For selected timeseries (timeseries already part of the monitoring network) it is easiest to develop if they are all loaded in advance. If this is too heavy we must add pagination to the right side column. But that is also the case for the current design if I understand correctly

**Comment by hoanphungt** [comment]
I think the slush bucket could be useful in this case. In my opinion, it would be nice to be able to search/add/remove timeseries from monitoring network all in one place. 

Compared to building a new modal for only deleting the timeseries and being able to reuse the existing slush bucket field (and improve upon it) to do all 3 things together (search/add/remove), I would definitely prefer the 2nd solution. 

I can pick this issue up to investigate further the use of the slush bucket in this case if performance wise, it is ok to handle large dataset.

**Comment by lexvand** [comment]
Okay, let's try to realize this slush bucket solution.

For reference: Currently on production the organisation Zuiderzeeland has most timeseries, around 180k. The largest monitoring network contains 5k timeseries.

Relevant search options:
- Code
- Name
- Location_code
- Location_name
- Parameter

**Comment by LirryPinter** [comment]
@hoanphungt @lexvand @Derryrover 

Yes lets test this approach!

**Comment by hoanphungt** [comment]
@lexvand @Derryrover @LirryPinter 
I am trying to use the SlushBucket for this modal and realise that there are quite a number of problems:

1. First of all is that the the timeseries sub-endpoint (e.g. https://nxt3.staging.lizard.net/api/v4/monitoringnetworks/0026cda8-5497-4496-9a76-1c18b81c895c/timeseries/) supports 2 different types of requests: `POST` and `DELETE`. So adding a new timeseries and removing an existing timeseries from the monitoring network require 2 different actions. This means I need to keep track of what timeseries have been removed and what have been added compared to the original list of timeseries in order to make the right `POST` and `DELETE` requests when the form is submitted.

2. The second problem is the large dataset of both available timeseries for an organisation and selected timeseries of a monitoring network. This monitoring network (https://nxt3.staging.lizard.net/api/v4/monitoringnetworks/0026cda8-5497-4496-9a76-1c18b81c895c/) contains more than 5K timeseries already while Nelen & Schuurmans has around 18K timeseries to select from (as Lex mentioned Zuiderzeeland has around 180K timeseries). It means we would need to develop the SlushBucket to use pagination in both buckets and potentially need to add a search bar to the right bucket too. Also, due to the large dataset, we need to add a loading state to both buckets.

In conclusion, I think we would need to spend considerable amount of time in order to make the SlushBucket work adequately for this modal. Meanwhile, the current design of the modal also has the problem that the list of timeseries in the monitoring network is also not paginated but this can be added more easily. So in my opinion, it's better to develop with the current design by Lirry.

What do you guys think?

**Comment by Derryrover** [comment]
@hoanphungt 
@LirryPinter 

Hi Hoan and Lirry,
Good we are picking this up!

@hoanphungt : Given the constraints with large datasets it might be worth trying to use the TableStateContainer (or an adapted variaty of this) for both left and right side of the slushbucket. This way we could reuse search and pagination.

@LirryPinter : Maybe you could draft some wireframes for the solution. The problematic part is that both side of the slushbucket probably need pagination and filter.

kr Tom

**Comment by hoanphungt** [comment]
@Derryrover 
Hi Tom,

Could you review this PR: https://github.com/nens/lizard-management-client/issues/802?

Kr, Hoan

**Comment by Derryrover** [comment]
@hoanphungt 

Hi Hoan,

Looks good!
I see you also fixed some typing for components like pagnation.
Very good

Ready to merge

kr Tom
